extra_geom/detectors.py

```diff
@@ -857,6 +857,43 @@ def from_simple_description(cls, pixel_size: float, slow_pixels: int, fast_pixel
 
         return geom
 
+    def inspect(self, axis_units='px', frontview=True, aspect='auto'):
+        """Plot the 2D layout of this detector geometry.
+
+        Returns a matplotlib Axes object.
+
+        Parameters
+        ----------
+
+        axis_units : str
+          Show the detector scale in pixels ('px') or metres ('m').
+        frontview : bool
+          If True (the default), x increases to the left, as if you were looking
+          along the beam. False gives a 'looking into the beam' view.
+        aspect : str, float
+          Set the aspect ratio of the plot, possible values:
+            * 'auto' (default): automatic; fill the position rectangle with data
+            * 'equal': same scaling from data to plot units for x and y
+            * a number: a figure will be stretched such that the height is num times the width.
+              aspect=1 is the same as 'equal'.
+        """
+        ax = super().inspect(axis_units=axis_units, frontview=frontview)
+
+        ax.set_title(f"{self.detector_type_name} geometry ({self.filename})")
+
+        scale = self._get_plot_scale_factor(axis_units)
+
+        # Label modules and tiles
+        for ch, module in enumerate(self.modules):
+            if self.n_tiles_per_module > 1:
+                for t in [0, self.n_tiles_per_module - 1]:
+                    cx, cy, _ = module[t].centre() * scale
+                    ax.text(cx, cy, 'M{}T{}'.format(ch + 1, t + 1),
```

Comment:
Up to you whether it counts from 1 or 0. For the Q1M1 numbers, starting at 1 seems to be the norm, but when we're just counting modules, there are different conventions in different places.

Reply:
That's fine - I think either way it's going to be what some people expect and not what others expect. I think I'd go with starting from zero. We can always add an option later if people complain enough.